app/services/oidc_jwt_token_validate.rb

```diff
@@ -0,0 +1,45 @@
+class OidcJwtTokenValidate
+  attr_reader :decoded_token
+  delegate :logger, to: :Rails
+
+  def initialize(jwt_token)
+    @jwt_token = jwt_token
+  end
+
+  def valid_jwt?
+    # OpenSSL#set_key method does not support ruby version < 2.4.0, apparently the JWT gem uses
+    # OpenSSL#set_key method for all ruby version. We must remove this condition once new version
+    # of the JWT(2.2.2) is released.
+    unless OpenSSL::PKey::RSA.new.respond_to?(:set_key)
+      Foreman::Logging.logger('notifications').error "SSO feature is not available for Ruby < 2.4.0"
```

Comment:
Added error logging in case of Ruby < 2.4.0. Here is an example log:
 
```
2019-10-04T15:41:12 [E|not|267ad15b] SSO feature is not available for Ruby < 2.4.0
2019-10-04T15:41:12 [E|app|267ad15b] JWT SSO: Failed to decode JWT.
2019-10-04T15:41:12 [W|app|267ad15b] SSO failed
```